scripts/cfos/make_dataset_data_tables.py

```diff
@@ -143,57 +108,98 @@ def load_excel_input(excel, expected_dataset_cols):
-    parser.add_argument('-w', '--workspace', required=False, type=str, help='Terra workspace name.')
-    parser.add_argument('-v', '--validate', required=False, action='store_true', help='Set parameter to run only validation on input excel file.')
-    parser.add_argument('-u', '--upload', required=False, action='store_true', help='Set parameter to upload data table files to workspace. default = no upload')
+    # REK TODO: update so JSON is assumed to live in an internal directory location and not as an input file. 
+    # Goal: allow qualified people to make easy updates to schemas, but don't encourage random users to play with it.
+
+    parser.add_argument(
+        '-d', 
+        '--dataset_name', 
+        required=False,
+        type=str, 
+        help='Dataset type to apply validations and data table structure. ex. proteomics, atac-seq, transcriptomics, singe-cell'
+    )
+    parser.add_argument(
+        '-x', 
+        '--excel', 
+        required=False,
+        type=str, 
+        help='CFoS data excel file (.xlsx) - based on CFoS data intake template file.'
+    )
+    parser.add_argument(
+        '-p', 
+        '--project', 
+        required=False, 
+        type=str, 
+        help='Terra workspace project (namespace).'
+    )
+    parser.add_argument(
+        '-w', 
+        '--workspace', 
+        required=False, 
+        type=str, 
+        help='Terra workspace name.'
+    )
+    parser.add_argument(
+        '-v', 
+        '--validate', 
+        required=False, 
+        action='store_true', 
+        help='Set parameter to run only validation on input excel file.'
+    )
+    parser.add_argument(
+        '-u', 
+        '--upload', 
+        required=False, 
+        action='store_true', 
+        help='Set parameter to upload data table files to workspace. default = no upload'
+    )
 
     args = parser.parse_args()
+    schema_json = 'dataset_tables_schema.json'
+
+    # TODO: Validate that schema exists; error and quit if not. Store validation code in new requirements or file_paths validation file.
+    # ask John - why write separate validation? Why not just use a Try block? 
+    # https://therenegadecoder.com/code/how-to-check-if-a-file-exists-in-python/#check-if-a-file-exists-with-a-try-block indicates try block is more robust.
 
     # parse schema using dataset name to get list of expected columns
-    expected_dataset_cols, dataset_table_names, schema_dict = get_expected_columns_list(args.schema, args.dataset_name)
+    schema_dict, column_dict = parse_config_file(schema_json=schema_json)
     # load excel to dataframe validating to check if all expected columns present
-    dataset_metadata = load_excel_input(args.excel, expected_dataset_cols)
+    dataset_metadata = load_excel_input(args.excel, list(column_dict.keys()))
 
     # if validation flag, only validate
     if args.validate:
         validated_dataset = validate_dataset(dataset_metadata)
```

Comment:
you need to update this invocation of the function to include the additional new input args (schema_dict, tsv_filenames_list)

Reply:
I'm only seeing two on my version of the code. Maybe mine is a later version. Can you take a look again once I commit and see if this is resolved?

Reply:
There was definitely something missing from the validate function, so maybe this was in reference to that? I fixed it. I probably messed it up during my last set of edits. :(